modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1121,6 +1122,24 @@ const DOMUtils = (doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
     return state;
   };
 
+  const getContentEditableRoot = (node: Node) => {
+    const root = getRoot();
+    let editableRoot: HTMLElement | undefined;
+    // Get all parents until we hit a non editable parent or the root
+    let parent: Node | null = node;
+    while (parent !== root && parent) {
+      if (getContentEditable(parent) === 'false') {
+        // node is within contenteditable="false" element, return null
+        return null;
+      } else if (getContentEditable(parent) === 'true') {
+        editableRoot = parent as HTMLElement;
+      }
+      parent = parent.parentNode;
+    }
+    // return the ancestor contenteditable="true" or root
+    return parent !== root && editableRoot ? editableRoot : root;
```

Comment:
This here is another issue if you're making this a public API... this doesn't make sense to be returning the root here, because it then would be inconsistent with the existing `getContentEditableParent` which returns `null` if the root is what's matched.